src/main/java/net/neoforged/moddevgradle/internal/IdeIntegration.java

```diff
@@ -0,0 +1,108 @@
+package net.neoforged.moddevgradle.internal;
+
+import net.neoforged.moddevgradle.dsl.ModModel;
+import net.neoforged.moddevgradle.dsl.RunModel;
+import net.neoforged.moddevgradle.internal.utils.IdeDetection;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.file.Directory;
+import org.gradle.api.file.RegularFile;
+import org.gradle.api.provider.Property;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.SetProperty;
+import org.gradle.api.tasks.TaskProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.Map;
+import java.util.WeakHashMap;
+
+sealed abstract class IdeIntegration permits IntelliJIntegration, EclipseIntegration, NoIdeIntegration {
+    private static final Logger LOG = LoggerFactory.getLogger(IdeIntegration.class);
+
+    private static final WeakHashMap<Project, IdeIntegration> CACHE = new WeakHashMap<>();
```

Comment:
You should make this an identity hash map because
1. The values refer to the key so they will never get cleaned anyway.
2. Since the `IdeIntegration` registers a new task in its constructor, we don't want to create a second one for the same project.

Reply:
Projects should not be used as keys in an `IdentityHashMap` in general as they are not safe to compare by identity -- two `Project` objects can be different by identity but be the same gradle project (`MutableStateAccessAwareProject`, for instance, comes to mind).

Reply:
Furthermore, the weak reference here actually seems like it'd be _too_ strong, really -- you don't want to keep a reference to a `Project` around indefinitely, as gradle should be able to toss it out after the configuration phase and have everything attached -- extensions, etc. -- be GCed, recovering a not-insubstantial amount of memory. I would say that this should be made a `WeakHashMap<Project, WeahReference<IdeIntegration>>`, and then keep the integration not-GC-ed by making it an extension on the project, or something -- though I wonder why it's not kept around via an just extension to begin with? Otherwise you would undo some of the benefits conf cache should be able to bring